Automattic/jetpack
# pro-status: Use feature checks for VPScan and VPBackup

#### Changes proposed in this Pull Request:

* In pro-status, change VaultPress Scan and Backup checks to use feature checks instead of plan checks
  * Instead of using the logic that a free plan means they haven't purchased VaultPress scan and VaultPress backups, use the feature system to check if their plan provides those features
    * Our definition of `VAULTPRESS_BACKUPS` is wrong and should actually be `JETPACK_PERSONAL_AND_HIGHER`. This will need to be done in another PR. @obenland has seen another mention of this discrepancy as well.
      * Done here: D79837-code
      * Other reference is [here](https://github.com/Automattic/jetpack/pull/24121/files#r861998686)
  * Pro-status controls the top right of each card in the Jetpack Dashboard
    * For this section, it can display: Nothing, "Set Up" buttons inviting you to start using a feature you've purchased that you don't use, or a "Scan OK" or "Threats Detected" type of notice (None of which are in my screenshot).
  
![2022-04-29_14-11](https://user-images.githubusercontent.com/937354/166034030-dcd7943e-0125-470f-ad20-87c3521d33df.png)


#### Jetpack product discussion


#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Currently hazy. Roughly, we will need to install/enable VaultPress (a standalone plugin) and make sure that on the top right of the cards in the Jetpack dashboard, you are invited to "Set Up" scan and backups if you have purchased them but have not yet enabled them. After you enable scan, the top right of the scan badge should show scan status.
  * If you have not purchased them, (you are on Jetpack free), then you should not see the SetUp buttons.
  * Behavior should be the same before and after.
  * I will be giving this a try and updating the PR as I find anything.

There may be another issue - not related - where the Jetpack Dashboard does not display properly when VaultPress is enabled. Some cards were always blinking in and out and my browser was making endless network requests. This should be handled in another PR if it is a real issue.
  * PR to fix this: #24194
